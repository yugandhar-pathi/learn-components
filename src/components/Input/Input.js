import PropTypes from 'prop-types'
import { LabelStyle, InputStyle } from './Input.styles'

export const Input = ({text, styleType, placeHolder, dataTestId, type}) =>{

    return(
        <div data-testid={dataTestId}>
        <LabelStyle>{text}</LabelStyle>
        <InputStyle styleType={styleType} type={type} placeholder={placeHolder}/>
        </div>
    )
}

Input.propTypes = {
 styleType : PropTypes.oneOf(["basic", "outlined", "filled"]),
 placeHolder : PropTypes.string.isRequired,
 text : PropTypes.string.isRequired,
 dataTestId: PropTypes.string.isRequired,
 type : PropTypes.string.isRequired
}

Input.defaultProps = {
    styleType: "basic" ,
    placeHolder: "Enter Your Input" 
}