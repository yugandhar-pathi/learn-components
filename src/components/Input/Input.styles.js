import styled from 'styled-components'

export const LabelStyle = styled.label`
 color : black;
 font-size : 20px;
`
export const InputStyle = styled.input`
  width: 25%;
  height: 40px;
  border:${(props) => props.styleType === "basic" ? "1px solid black" : ""};
  border:${(props) => props.styleType === "outlined" ? "2px solid black" : ""};
  color:${(props) => props.styleType === "outlined" ? "blue" : ""};
  background-color:${(props) => props.styleType === "filled" ? "gray" : ""};
  color:${(props) => props.styleType === "filled" ? "green" : ""};
`