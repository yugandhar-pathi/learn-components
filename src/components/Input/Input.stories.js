import React from 'react';
import { Input } from './Input';

export default{
    component : Input,
}

export const basic = () => <Input text="Enter User Name Basic" basic/>
export const outlined = () => <Input text="Enter User Name outlined" outlined/>
export const filled = () => <Input text="Enter User Name Filled" filled/>