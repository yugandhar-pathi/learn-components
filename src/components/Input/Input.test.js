import {render, screen} from '@testing-library/react';
import {Input} from './Input'

test("Input component", ()=>{
    render(<Input text="my text" dataTestId='input-id' placeHolder="my placeholder" styleType="basic"/>)
    expect(screen.getByTestId("input-id").textContent).toBe("my text")
})

test("snapshot for Input", ()=>{
    render(<Input text="my text" dataTestId='input-id' placeHolder="my placeholder" styleType="basic"/>)
    expect(screen.getByTestId("input-id")).toMatchSnapshot()
})