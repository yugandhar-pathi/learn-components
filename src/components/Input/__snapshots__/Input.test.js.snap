// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for Input 1`] = `
<div
  data-testid="input-id"
>
  <label
    class="sc-bczRLJ kZPjBP"
  >
    my text
  </label>
  <input
    class="sc-gsnTZi fYkahX"
    placeholder="my placeholder"
  />
</div>
`;
